src/main/kotlin/io/spine/publishing/gradle/InterdependentLibraries.kt

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2020, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.publishing.gradle
+
+/**
+ * A number of libraries that have dependencies between them.
+ *
+ * Allows to [update and publish new versions][InterdependentLibraries.publish] in a
+ * dependency-based order.
+ */
+class InterdependentLibraries(private val ordering: Ordering) {
+
+    /**
+     * Goes through the libraries one by one, updating their version, building and publishing
+     * new versions them to the remote artifact repositories.
+     *
+     * First, updates the versions of all libraries. Then, starts building and publishing the
+     * libraries in a dependency-based order, e.g:
+     *
+     * library A depends on library B and C. Library B depends on library C.
+     * In such a configuration, the following happens:
+     *
+     * 1) library C is built;
+     * 2) library C is published to the local Maven repo, so that it can be used for further builds;
+     * 3) library B is built;
+     * 4) library B is published to the local Maven repo;
+     * 5) library A is built;
+     * 6) library A is published to the local Maven repo;
+     * 7) library A, B and C are published to the remote artifact repository, as at this point
```

Comment:
Aren't you tired of typing the word "library"? Why "A", "B" and "C" are not enough?